### DISCUSSION TITLE
Flask build error

### DISCUSSION THREAD

**Discussion by xanmankey** [vercel/vercel#7714] [Help]
I asked a similar question in a python-javabridge forum, but I figured I would follow it up here as well: from what I understand, numpy is a build-dependency of python-javabridge, and my flask app depends on python-javabridge, so when I try to build it using vercel, the command fails with a ModuleNotFound: numpy error. In particular, I'm wondering if there is a way I can install numpy first, and then install from my requirements.txt file for the vercel build process (as I believe that would circumvent the issue; so far I haven't seen much information regarding this concept though, so I figured I would ask myself, see if that would be a feasible solution or not). The compiler error and my vercel.json can be found below, please let me know if you need any more information regarding the issue or have any suggestions.

Full compiler output (minus the cloning at the beginning):
> 10:27:42.159 | Warning: Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings
10:27:42.480 | Installing build runtime...
10:27:44.747 | Build runtime installed: 2.266s
10:27:45.452 | Looking up build cache...
10:27:45.748 | Build Cache not found
10:27:45.883 | Installing required dependencies...
10:28:01.839 | Failed to run "pip3.9 install --disable-pip-version-check --target . --upgrade -r /vercel/path0/requirements.txt"
10:28:01.844 | Error: Command failed: pip3.9 install --disable-pip-version-check --target . --upgrade -r /vercel/path0/requirements.txt
10:28:01.844 | error: subprocess-exited-with-error
10:28:01.844 |  
10:28:01.845 | × python setup.py egg_info did not run successfully.
10:28:01.845 | │ exit code: 1
10:28:01.845 | ╰─> [8 lines of output]
10:28:01.845 | Traceback (most recent call last):
10:28:01.845 | File "<string>", line 2, in <module>
10:28:01.845 | File "<pip-setuptools-caller>", line 34, in <module>
10:28:01.846 | File "/tmp/pip-install-9g1f79tv/python-javabridge_fc461e205f7f4afc87c818441711f35e/setup.py", line 412, in <module>
10:28:01.846 | ext_modules=ext_modules(),
10:28:01.846 | File "/tmp/pip-install-9g1f79tv/python-javabridge_fc461e205f7f4afc87c818441711f35e/setup.py", line 100, in ext_modules
10:28:01.846 | from numpy import get_include
10:28:01.846 | ModuleNotFoundError: No module named 'numpy'
10:28:01.846 | [end of output]
10:28:01.846 |  
10:28:01.847 | note: This error originates from a subprocess, and is likely not a problem with pip.
10:28:01.847 | error: metadata-generation-failed
10:28:01.847 |  
10:28:01.847 | × Encountered error while generating package metadata.
10:28:01.847 | ╰─> See above for output.
10:28:01.847 |  
10:28:01.847 | note: This is an issue with the package mentioned above, not pip.
10:28:01.848 | hint: See above for details.
10:28:01.848 |  
10:28:01.848 | Collecting apscheduler==3.9.1
10:28:01.848 | Downloading APScheduler-3.9.1-py2.py3-none-any.whl (59 kB)
10:28:01.848 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 KB 2.9 MB/s eta 0:00:00
10:28:01.848 | Collecting async-generator==1.10
10:28:01.850 | Downloading async_generator-1.10-py3-none-any.whl (18 kB)
10:28:01.850 | Collecting attrs==21.4.0
10:28:01.850 | Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)
10:28:01.850 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.6/60.6 KB 2.2 MB/s eta 0:00:00
10:28:01.850 | Collecting backports.zoneinfo==0.2.1
10:28:01.850 | Downloading backports.zoneinfo-0.2.1.tar.gz (74 kB)
10:28:01.851 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.1/74.1 KB 12.7 MB/s eta 0:00:00
10:28:01.851 | Installing build dependencies: started
10:28:01.851 | Installing build dependencies: finished with status 'done'
10:28:01.851 | Getting requirements to build wheel: started
10:28:01.851 | Getting requirements to build wheel: finished with status 'done'
10:28:01.851 | Preparing metadata (pyproject.toml): started
10:28:01.851 | Preparing metadata (pyproject.toml): finished with status 'done'
10:28:01.852 | Collecting beautifulsoup4==4.11.1
10:28:01.852 | Downloading beautifulsoup4-4.11.1-py3-none-any.whl (128 kB)
10:28:01.852 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 KB 19.2 MB/s eta 0:00:00
10:28:01.852 | Collecting certifi==2021.10.8
10:28:01.852 | Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
10:28:01.852 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.2/149.2 KB 25.1 MB/s eta 0:00:00
10:28:01.852 | Collecting cffi==1.15.0
10:28:01.852 | Downloading cffi-1.15.0-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (444 kB)
10:28:01.853 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 444.3/444.3 KB 23.4 MB/s eta 0:00:00
10:28:01.853 | Collecting charset-normalizer==2.0.12
10:28:01.853 | Downloading charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
10:28:01.853 | Collecting click==8.1.2
10:28:01.853 | Downloading click-8.1.2-py3-none-any.whl (96 kB)
10:28:01.853 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 KB 17.0 MB/s eta 0:00:00
10:28:01.853 | Collecting colorama==0.4.4
10:28:01.854 | Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
10:28:01.854 | Collecting cryptography==36.0.2
10:28:01.854 | Downloading cryptography-36.0.2-cp36-abi3-manylinux_2_24_x86_64.whl (3.6 MB)
10:28:01.854 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.6/3.6 MB 65.8 MB/s eta 0:00:00
10:28:01.854 | Collecting cssutils==2.4.0
10:28:01.854 | Downloading cssutils-2.4.0-py3-none-any.whl (404 kB)
10:28:01.854 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 405.0/405.0 KB 47.7 MB/s eta 0:00:00
10:28:01.854 | Collecting decorator==5.1.1
10:28:01.855 | Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)
10:28:01.855 | Collecting filelock==3.6.0
10:28:01.855 | Downloading filelock-3.6.0-py3-none-any.whl (10.0 kB)
10:28:01.855 | Collecting filetype==1.0.13
10:28:01.855 | Downloading filetype-1.0.13-py2.py3-none-any.whl (17 kB)
10:28:01.855 | Collecting fishingforphish==0.3.0
10:28:01.855 | Downloading FishingForPhish-0.3.0-py3-none-any.whl (40 kB)
10:28:01.855 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.9/40.9 KB 8.1 MB/s eta 0:00:00
10:28:01.855 | Collecting flask-apscheduler==1.12.3
10:28:01.855 | Downloading Flask-APScheduler-1.12.3.tar.gz (12 kB)
10:28:01.856 | Preparing metadata (setup.py): started
10:28:01.856 | Preparing metadata (setup.py): finished with status 'done'
10:28:01.856 | Collecting flask==2.1.1
10:28:01.857 | Downloading Flask-2.1.1-py3-none-any.whl (95 kB)
10:28:01.857 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 KB 18.2 MB/s eta 0:00:00
10:28:01.857 | Collecting h11==0.13.0
10:28:01.857 | Downloading h11-0.13.0-py3-none-any.whl (58 kB)
10:28:01.857 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.2/58.2 KB 10.2 MB/s eta 0:00:00
10:28:01.858 | Collecting idna==3.3
10:28:01.858 | Downloading idna-3.3-py3-none-any.whl (61 kB)
10:28:01.858 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 KB 11.5 MB/s eta 0:00:00
10:28:01.858 | Collecting imagehash==4.2.1
10:28:01.858 | Downloading ImageHash-4.2.1.tar.gz (812 kB)
10:28:01.858 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 812.6/812.6 KB 61.1 MB/s eta 0:00:00
10:28:01.858 | Preparing metadata (setup.py): started
10:28:01.858 | Preparing metadata (setup.py): finished with status 'done'
10:28:01.858 | Collecting importlib-metadata==4.11.3
10:28:01.858 | Downloading importlib_metadata-4.11.3-py3-none-any.whl (18 kB)
10:28:01.859 | Collecting ipy==1.01
10:28:01.859 | Downloading IPy-1.01.tar.gz (33 kB)
10:28:01.859 | Preparing metadata (setup.py): started
10:28:01.859 | Preparing metadata (setup.py): finished with status 'done'
10:28:01.859 | Collecting itsdangerous==2.1.2
10:28:01.859 | Downloading itsdangerous-2.1.2-py3-none-any.whl (15 kB)
10:28:01.859 | Collecting jinja2==3.1.1
10:28:01.859 | Downloading Jinja2-3.1.1-py3-none-any.whl (132 kB)
10:28:01.859 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.6/132.6 KB 24.6 MB/s eta 0:00:00
10:28:01.859 | Collecting markupsafe==2.1.1
10:28:01.863 | Downloading MarkupSafe-2.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
10:28:01.863 | Collecting numpy==1.22.3
10:28:01.863 | Downloading numpy-1.22.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
10:28:01.863 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.8/16.8 MB 66.4 MB/s eta 0:00:00
10:28:01.863 | Collecting outcome==1.1.0
10:28:01.864 | Downloading outcome-1.1.0-py2.py3-none-any.whl (9.7 kB)
10:28:01.864 | Collecting packaging==21.3
10:28:01.864 | Downloading packaging-21.3-py3-none-any.whl (40 kB)
10:28:01.864 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 KB 6.9 MB/s eta 0:00:00
10:28:01.864 | Collecting pillow==9.1.0
10:28:01.864 | Downloading Pillow-9.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
10:28:01.864 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 62.9 MB/s eta 0:00:00
10:28:01.864 | Collecting pycparser==2.21
10:28:01.865 | Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
10:28:01.865 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 KB 22.2 MB/s eta 0:00:00
10:28:01.866 | Collecting pymongo==4.1.1
10:28:01.866 | Downloading pymongo-4.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (471 kB)
10:28:01.866 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 471.3/471.3 KB 46.8 MB/s eta 0:00:00
10:28:01.866 | Collecting pyopenssl==22.0.0
10:28:01.866 | Downloading pyOpenSSL-22.0.0-py2.py3-none-any.whl (55 kB)
10:28:01.867 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.8/55.8 KB 10.9 MB/s eta 0:00:00
10:28:01.867 | Collecting pyparsing==3.0.8
10:28:01.867 | Downloading pyparsing-3.0.8-py3-none-any.whl (98 kB)
10:28:01.867 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.5/98.5 KB 19.2 MB/s eta 0:00:00
10:28:01.867 | Collecting pyshorteners==1.0.1
10:28:01.867 | Downloading pyshorteners-1.0.1.tar.gz (10.0 kB)
10:28:01.867 | Preparing metadata (setup.py): started
10:28:01.867 | Preparing metadata (setup.py): finished with status 'done'
10:28:01.868 | Collecting pysocks==1.7.1
10:28:01.868 | Downloading PySocks-1.7.1-py3-none-any.whl (16 kB)
10:28:01.868 | Collecting python-dateutil==2.8.2
10:28:01.868 | Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
10:28:01.868 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 KB 34.0 MB/s eta 0:00:00
10:28:01.868 | Collecting python-dotenv==0.20.0
10:28:01.868 | Downloading python_dotenv-0.20.0-py3-none-any.whl (17 kB)
10:28:01.868 | Collecting python-javabridge==4.0.3
10:28:01.868 | Downloading python-javabridge-4.0.3.tar.gz (1.3 MB)
10:28:01.868 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 60.1 MB/s eta 0:00:00
10:28:01.869 | Preparing metadata (setup.py): started
10:28:01.869 | Preparing metadata (setup.py): finished with status 'error'
10:28:01.869 |  
10:28:01.869 | at makeError (/vercel/e6c54d1441fc48ab/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:283:9)
10:28:01.869 | at /vercel/e6c54d1441fc48ab/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:387:16
10:28:01.869 | at processTicksAndRejections (internal/process/task_queues.js:95:5)
10:28:01.869 | at async pipInstall (/vercel/e6c54d1441fc48ab/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3777:9)
10:28:01.869 | at async Object.installRequirementsFile (/vercel/e6c54d1441fc48ab/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3809:5)
10:28:01.869 | at async Object.build (/vercel/e6c54d1441fc48ab/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3653:9) {
10:28:01.869 | code: 1,
10:28:01.870 | stdout: 'Collecting apscheduler==3.9.1\n' +
10:28:01.870 | '  Downloading APScheduler-3.9.1-py2.py3-none-any.whl (59 kB)\n' +
10:28:01.870 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 KB 2.9 MB/s eta 0:00:00\n' +
10:28:01.870 | 'Collecting async-generator==1.10\n' +
10:28:01.870 | '  Downloading async_generator-1.10-py3-none-any.whl (18 kB)\n' +
10:28:01.870 | 'Collecting attrs==21.4.0\n' +
10:28:01.870 | '  Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)\n' +
10:28:01.870 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.6/60.6 KB 2.2 MB/s eta 0:00:00\n' +
10:28:01.870 | 'Collecting backports.zoneinfo==0.2.1\n' +
10:28:01.870 | '  Downloading backports.zoneinfo-0.2.1.tar.gz (74 kB)\n' +
10:28:01.871 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.1/74.1 KB 12.7 MB/s eta 0:00:00\n' +
10:28:01.871 | '  Installing build dependencies: started\n' +
10:28:01.871 | "  Installing build dependencies: finished with status 'done'\n" +
10:28:01.871 | '  Getting requirements to build wheel: started\n' +
10:28:01.871 | "  Getting requirements to build wheel: finished with status 'done'\n" +
10:28:01.871 | '  Preparing metadata (pyproject.toml): started\n' +
10:28:01.871 | "  Preparing metadata (pyproject.toml): finished with status 'done'\n" +
10:28:01.871 | 'Collecting beautifulsoup4==4.11.1\n' +
10:28:01.871 | '  Downloading beautifulsoup4-4.11.1-py3-none-any.whl (128 kB)\n' +
10:28:01.871 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 KB 19.2 MB/s eta 0:00:00\n' +
10:28:01.871 | 'Collecting certifi==2021.10.8\n' +
10:28:01.871 | '  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n' +
10:28:01.872 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 149.2/149.2 KB 25.1 MB/s eta 0:00:00\n' +
10:28:01.872 | 'Collecting cffi==1.15.0\n' +
10:28:01.872 | '  Downloading cffi-1.15.0-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (444 kB)\n' +
10:28:01.872 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 444.3/444.3 KB 23.4 MB/s eta 0:00:00\n' +
10:28:01.872 | 'Collecting charset-normalizer==2.0.12\n' +
10:28:01.872 | '  Downloading charset_normalizer-2.0.12-py3-none-any.whl (39 kB)\n' +
10:28:01.872 | 'Collecting click==8.1.2\n' +
10:28:01.872 | '  Downloading click-8.1.2-py3-none-any.whl (96 kB)\n' +
10:28:01.872 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 KB 17.0 MB/s eta 0:00:00\n' +
10:28:01.872 | 'Collecting colorama==0.4.4\n' +
10:28:01.872 | '  Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)\n' +
10:28:01.872 | 'Collecting cryptography==36.0.2\n' +
10:28:01.872 | '  Downloading cryptography-36.0.2-cp36-abi3-manylinux_2_24_x86_64.whl (3.6 MB)\n' +
10:28:01.873 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.6/3.6 MB 65.8 MB/s eta 0:00:00\n' +
10:28:01.873 | 'Collecting cssutils==2.4.0\n' +
10:28:01.873 | '  Downloading cssutils-2.4.0-py3-none-any.whl (404 kB)\n' +
10:28:01.873 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 405.0/405.0 KB 47.7 MB/s eta 0:00:00\n' +
10:28:01.873 | 'Collecting decorator==5.1.1\n' +
10:28:01.873 | '  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)\n' +
10:28:01.873 | 'Collecting filelock==3.6.0\n' +
10:28:01.873 | '  Downloading filelock-3.6.0-py3-none-any.whl (10.0 kB)\n' +
10:28:01.873 | 'Collecting filetype==1.0.13\n' +
10:28:01.873 | '  Downloading filetype-1.0.13-py2.py3-none-any.whl (17 kB)\n' +
10:28:01.873 | 'Collecting fishingforphish==0.3.0\n' +
10:28:01.874 | '  Downloading FishingForPhish-0.3.0-py3-none-any.whl (40 kB)\n' +
10:28:01.874 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.9/40.9 KB 8.1 MB/s eta 0:00:00\n' +
10:28:01.874 | 'Collecting flask-apscheduler==1.12.3\n' +
10:28:01.874 | '  Downloading Flask-APScheduler-1.12.3.tar.gz (12 kB)\n' +
10:28:01.874 | '  Preparing metadata (setup.py): started\n' +
10:28:01.874 | "  Preparing metadata (setup.py): finished with status 'done'\n" +
10:28:01.874 | 'Collecting flask==2.1.1\n' +
10:28:01.874 | '  Downloading Flask-2.1.1-py3-none-any.whl (95 kB)\n' +
10:28:01.874 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 95.2/95.2 KB 18.2 MB/s eta 0:00:00\n' +
10:28:01.874 | 'Collecting h11==0.13.0\n' +
10:28:01.874 | '  Downloading h11-0.13.0-py3-none-any.whl (58 kB)\n' +
10:28:01.874 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.2/58.2 KB 10.2 MB/s eta 0:00:00\n' +
10:28:01.874 | 'Collecting idna==3.3\n' +
10:28:01.874 | '  Downloading idna-3.3-py3-none-any.whl (61 kB)\n' +
10:28:01.875 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 KB 11.5 MB/s eta 0:00:00\n' +
10:28:01.875 | 'Collecting imagehash==4.2.1\n' +
10:28:01.875 | '  Downloading ImageHash-4.2.1.tar.gz (812 kB)\n' +
10:28:01.875 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 812.6/812.6 KB 61.1 MB/s eta 0:00:00\n' +
10:28:01.875 | '  Preparing metadata (setup.py): started\n' +
10:28:01.875 | "  Preparing metadata (setup.py): finished with status 'done'\n" +
10:28:01.875 | 'Collecting importlib-metadata==4.11.3\n' +
10:28:01.875 | '  Downloading importlib_metadata-4.11.3-py3-none-any.whl (18 kB)\n' +
10:28:01.875 | 'Collecting ipy==1.01\n' +
10:28:01.875 | '  Downloading IPy-1.01.tar.gz (33 kB)\n' +
10:28:01.875 | '  Preparing metadata (setup.py): started\n' +
10:28:01.876 | "  Preparing metadata (setup.py): finished with status 'done'\n" +
10:28:01.876 | 'Collecting itsdangerous==2.1.2\n' +
10:28:01.876 | '  Downloading itsdangerous-2.1.2-py3-none-any.whl (15 kB)\n' +
10:28:01.876 | 'Collecting jinja2==3.1.1\n' +
10:28:01.876 | '  Downloading Jinja2-3.1.1-py3-none-any.whl (132 kB)\n' +
10:28:01.876 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.6/132.6 KB 24.6 MB/s eta 0:00:00\n' +
10:28:01.876 | 'Collecting markupsafe==2.1.1\n' +
10:28:01.876 | '  Downloading MarkupSafe-2.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)\n' +
10:28:01.876 | 'Collecting numpy==1.22.3\n' +
10:28:01.876 | '  Downloading numpy-1.22.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)\n' +
10:28:01.876 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.8/16.8 MB 66.4 MB/s eta 0:00:00\n' +
10:28:01.876 | 'Collecting outcome==1.1.0\n' +
10:28:01.877 | '  Downloading outcome-1.1.0-py2.py3-none-any.whl (9.7 kB)\n' +
10:28:01.877 | 'Collecting packaging==21.3\n' +
10:28:01.877 | '  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n' +
10:28:01.877 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 KB 6.9 MB/s eta 0:00:00\n' +
10:28:01.877 | 'Collecting pillow==9.1.0\n' +
10:28:01.877 | '  Downloading Pillow-9.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n' +
10:28:01.877 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 62.9 MB/s eta 0:00:00\n' +
10:28:01.877 | 'Collecting pycparser==2.21\n' +
10:28:01.877 | '  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)\n' +
10:28:01.877 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 KB 22.2 MB/s eta 0:00:00\n' +
10:28:01.878 | 'Collecting pymongo==4.1.1\n' +
10:28:01.878 | '  Downloading pymongo-4.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (471 kB)\n' +
10:28:01.878 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 471.3/471.3 KB 46.8 MB/s eta 0:00:00\n' +
10:28:01.878 | 'Collecting pyopenssl==22.0.0\n' +
10:28:01.878 | '  Downloading pyOpenSSL-22.0.0-py2.py3-none-any.whl (55 kB)\n' +
10:28:01.878 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.8/55.8 KB 10.9 MB/s eta 0:00:00\n' +
10:28:01.878 | 'Collecting pyparsing==3.0.8\n' +
10:28:01.878 | '  Downloading pyparsing-3.0.8-py3-none-any.whl (98 kB)\n' +
10:28:01.878 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.5/98.5 KB 19.2 MB/s eta 0:00:00\n' +
10:28:01.878 | 'Collecting pyshorteners==1.0.1\n' +
10:28:01.878 | '  Downloading pyshorteners-1.0.1.tar.gz (10.0 kB)\n' +
10:28:01.879 | '  Preparing metadata (setup.py): started\n' +
10:28:01.879 | "  Preparing metadata (setup.py): finished with status 'done'\n" +
10:28:01.879 | 'Collecting pysocks==1.7.1\n' +
10:28:01.879 | '  Downloading PySocks-1.7.1-py3-none-any.whl (16 kB)\n' +
10:28:01.879 | 'Collecting python-dateutil==2.8.2\n' +
10:28:01.879 | '  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n' +
10:28:01.879 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 KB 34.0 MB/s eta 0:00:00\n' +
10:28:01.879 | 'Collecting python-dotenv==0.20.0\n' +
10:28:01.879 | '  Downloading python_dotenv-0.20.0-py3-none-any.whl (17 kB)\n' +
10:28:01.879 | 'Collecting python-javabridge==4.0.3\n' +
10:28:01.879 | '  Downloading python-javabridge-4.0.3.tar.gz (1.3 MB)\n' +
10:28:01.880 | '     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 60.1 MB/s eta 0:00:00\n' +
10:28:01.880 | '  Preparing metadata (setup.py): started\n' +
10:28:01.880 | "  Preparing metadata (setup.py): finished with status 'error'\n",
10:28:01.880 | stderr: '  error: subprocess-exited-with-error\n' +
10:28:01.880 | '  \n' +
10:28:01.880 | '  × python setup.py egg_info did not run successfully.\n' +
10:28:01.880 | '  │ exit code: 1\n' +
10:28:01.880 | '  ╰─> [8 lines of output]\n' +
10:28:01.880 | '      Traceback (most recent call last):\n' +
10:28:01.880 | '        File "<string>", line 2, in <module>\n' +
10:28:01.880 | '        File "<pip-setuptools-caller>", line 34, in <module>\n' +
10:28:01.880 | '        File "/tmp/pip-install-9g1f79tv/python-javabridge_fc461e205f7f4afc87c818441711f35e/setup.py", line 412, in <module>\n' +
10:28:01.880 | '          ext_modules=ext_modules(),\n' +
10:28:01.881 | '        File "/tmp/pip-install-9g1f79tv/python-javabridge_fc461e205f7f4afc87c818441711f35e/setup.py", line 100, in ext_modules\n' +
10:28:01.881 | '          from numpy import get_include\n' +
10:28:01.881 | "      ModuleNotFoundError: No module named 'numpy'\n" +
10:28:01.881 | '      [end of output]\n' +
10:28:01.881 | '  \n' +
10:28:01.881 | '  note: This error originates from a subprocess, and is likely not a problem with pip.\n' +
10:28:01.881 | 'error: metadata-generation-failed\n' +
10:28:01.881 | '\n' +
10:28:01.881 | '× Encountered error while generating package metadata.\n' +
10:28:01.881 | '╰─> See above for output.\n' +
10:28:01.881 | '\n' +
10:28:01.881 | 'note: This is an issue with the package mentioned above, not pip.\n' +
10:28:01.881 | 'hint: See above for details.\n',
10:28:01.881 | failed: true,
10:28:01.881 | signal: null,
10:28:01.882 | cmd: 'pip3.9 install --disable-pip-version-check --target . --upgrade -r /vercel/path0/requirements.txt',
10:28:01.882 | timedOut: false,
10:28:01.882 | killed: false
10:28:01.882 | }

> {
    "version": 2,
    "builds": [
        {
            "src": "./index.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "/"
        }
    ]
}

**Comment by abdussalam02** [comment]
I am also stuck in this error

**Comment by amyegan** [comment]
This could be caused by a bad dependency version, as described in https://github.com/vercel/vercel/discussions/5403.

Does the error appear only on Vercel deployments, or is it also throwing this error if you run `vercel build` or `vercel dev` locally? If it succeeds in a local environment, [Build Output API](https://vercel.com/blog/build-output-api) could be a decent workaround to get you unstuck until we can figure out what's going wrong.
